source/shared/cpp/ObjectModel/MarkDownParser.cpp

```diff
@@ -0,0 +1,477 @@
+#include <vector>
+#include <iomanip>
+#include <iostream>
+#include "MarkDownParser.h"
+
+using namespace AdaptiveCards;
+
+MarkDownParser::MarkDownParser(const std::string &txt) : m_text(txt), m_curPos(m_text.begin())
+{
+}
+
+// adjust number of emphasis counts after maching is done
+int MarkDownParser::AdjustEmphasisCounts(int leftOver, std::list<Emphasis>::iterator left_itr, 
+        std::list<Emphasis>::iterator right_itr)
+{
+    int delimCnts = 0;
+    if (leftOver >= 0)
+    {
+        delimCnts = left_itr->m_emphCnts - leftOver;
+        left_itr->m_emphCnts = leftOver;
+        right_itr->m_emphCnts = 0;
+    }
+    else
+    {
+        delimCnts = left_itr->m_emphCnts;
+        right_itr->m_emphCnts = leftOver * (-1);
+        left_itr->m_emphCnts = 0;
+    }
+    return delimCnts;
+}
+
+// Following the rules speicified in CommonMark (http://spec.commonmark.org/0.27/)
+// It generally supports more stricker version of the rules
+// push left delims to stack, until matching right delim is found,
+// update emphasis counts and type to build string after search is complete
+std::string MarkDownParser::TransformToHtml()
+{
+    GenSymbolTable();
+    if (!m_leftLookUpTable.size() || !m_rightLookUpTable.size())
```

Comment:
.Size() returns an int. It's usually more readable to compare against 0 than to treat the result as a bool.

Since you don't really care about the actual size, but just want to check if the list is empty, you can use m_leftLookupTable.Empty()